language/move-vm/runtime/src/tracing.rs

```diff
@@ -1,6 +1,14 @@
 // Copyright (c) The Libra Core Contributors
 // SPDX-License-Identifier: Apache-2.0
 
+#[cfg(debug_assertions)]
```

Comment:
Should we make `debug_assertions` a feature for `move-vm-runtime` and `libra-vm` and make sure it is off by default so that they won't be activated by default?

Reply:
Not sure if I quite follow so I'd be interested in hearing more about your thoughts here. 

My thinking at least was that `debug_assertions` are automatically turned off in release builds, and these being around for debug builds are fine since it might just slow things down a very little bit (one extra Rust function call per instruction).

Reply:
That is indeed what's happening right now: release build of VM binary will not include those features whereas debug build will contain those tracing and debugging logic. That's the reason why `transaction-replay` get this feature without any changes made over there, which is really cool. What I'm suggestingis that instead of using `debug_assertions` flag, we can make those become a feature of `move_vm_runtime` that is switched automatically of in the release build, and we can choose if we need to enable it in the debug/test phase.

Reply:
I guess my main question here that I'm still wondering about is what adding the additional flag buys us. Having this switched on in debug builds should have essentially no impact, but, I worry that adding the additional flag adds additional complexity across the codebase since other crates that pull in the VM will needing to include this feature flag  to then pass down to the VM etc.  What are your thoughts on this @dariorussi and @vgao1996?

Reply:
this does not seem a critical aspect for now, or at least not in my opinion.
The different flag could allow us to build a release with VM debug capability if needed, so that would be the win for me. Not that we could not do it otherwise but creates a clear "feature" to depend on. 
However I am not clear what we need now, so this should be ok.
Thoughts?